easy-ui-react/src/ProductLayout/ProductLayoutSidebar.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactNode, useCallback } from "react";
+import { Overlay, useDialog, useModalOverlay } from "react-aria";
+import { classNames } from "../utilities/css";
+import { ScreenSizeRenderer } from "./ScreenSizeRenderer";
+import { useProductLayout } from "./context";
+
+import styles from "./ProductLayoutSidebar.module.scss";
+
+export type ProductLayoutSidebarProps = {
+  /**
+   * The content to render in the sidebar.
+   */
+  children: ReactNode;
+};
+
+export function ProductLayoutSidebar(props: ProductLayoutSidebarProps) {
+  const { sidebarTriggerState } = useProductLayout();
+  const handleScreenSizeChange = useCallback(
+    (isLargeScreen: boolean) => {
+      if (isLargeScreen) {
+        sidebarTriggerState.close();
+      }
+    },
+    [sidebarTriggerState],
+  );
+  return (
+    <ScreenSizeRenderer
+      onChange={handleScreenSizeChange}
+      renderOnLargeScreen={() => <LargeScreenSidebar {...props} />}
+      renderOnSmallScreen={() => <SmallScreenSidebar {...props} />}
+    />
```

Comment:
the sidebar is a `role="region"` on large screens and a `role="desktop"` on small screens. given the different underlying implementation, felt it easier to render as different trees